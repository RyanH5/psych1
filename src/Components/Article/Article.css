.article-link {
  color: black;
  font-size: 18px;
  font-weight: 600;
}

.single-article {
  background: #fff;
  padding: 10px;
  border-radius: 16px;
  border: 2px solid gray;
  margin-bottom: 10px;
}