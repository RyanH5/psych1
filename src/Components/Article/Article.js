import React from 'react';
import "./Article.css";

const Article = (props) => {

  return (
    <a 
      className="single-article"
      target="_blank" 
      href={props.url}
      >
      <div className="article-container">
      <span className="link-text">{props.title}</span>
      <img 
        className="news-image"
        src={props.urlToImage} 
      />
      <p>{props.description}</p>
      <h5 className="article-author">{props.author}</h5>
      <h6 className="article-source">{props.source.name}</h6>   
      </div> 
    </a> 
  )     
}

export default Article;