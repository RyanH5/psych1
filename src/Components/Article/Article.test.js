import React from 'react';
import { shallow } from 'enzyme';
import Article from './Article';

describe('Article', () => {
  it('should match snapshot', () => {
    let wrapper = shallow(<Article source={{name: 'jill'}}/>)

    expect(wrapper).toMatchSnapshot();
  })
})