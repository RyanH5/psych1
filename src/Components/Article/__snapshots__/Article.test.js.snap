// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Article
    source={
        Object {
            "name": "jill",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="article-container"
>
        <span
                className="link-text"
        />
        <img
                className="news-image"
                src={undefined}
        />
        <p />
        <h5
                className="article-author"
        />
        <h6
                className="article-source"
        >
                jill
        </h6>
</div>,
      "className": "single-article",
      "href": undefined,
      "target": "_blank",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="link-text"
/>,
          <img
            className="news-image"
            src={undefined}
/>,
          <p />,
          <h5
            className="article-author"
/>,
          <h6
            className="article-source"
>
            jill
</h6>,
        ],
        "className": "article-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "link-text",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "news-image",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "article-author",
          },
          "ref": null,
          "rendered": null,
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "jill",
            "className": "article-source",
          },
          "ref": null,
          "rendered": "jill",
          "type": "h6",
        },
      ],
      "type": "div",
    },
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="article-container"
>
          <span
                    className="link-text"
          />
          <img
                    className="news-image"
                    src={undefined}
          />
          <p />
          <h5
                    className="article-author"
          />
          <h6
                    className="article-source"
          >
                    jill
          </h6>
</div>,
        "className": "single-article",
        "href": undefined,
        "target": "_blank",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="link-text"
/>,
            <img
              className="news-image"
              src={undefined}
/>,
            <p />,
            <h5
              className="article-author"
/>,
            <h6
              className="article-source"
>
              jill
</h6>,
          ],
          "className": "article-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "link-text",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "news-image",
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "article-author",
            },
            "ref": null,
            "rendered": null,
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "jill",
              "className": "article-source",
            },
            "ref": null,
            "rendered": "jill",
            "type": "h6",
          },
        ],
        "type": "div",
      },
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
